dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.h

```diff
@@ -570,7 +570,7 @@ class OpenDDS_Rtps_Udp_Export RtpsUdpDataLink : public DataLink, public virtual
       , id_(id)
       , stopping_(false)
       , nackfrag_count_(0)
-      , preassociation_task_(link->reactor_task_->interceptor(), *this, &RtpsReader::send_preassociation_acknacks)
+      , preassociation_task_(new RtpsReader::Sporadic(link->reactor_task_->interceptor(), *this, &RtpsReader::send_preassociation_acknacks), keep_count())
```

Comment:
Use the `make_rch` helper (like `std::make_shared`)

Reply:
I originally had it that way, but unfortunately it wouldn't compile. I believe it was a `const` issue for the constructor... the macro assumes the input is const, but the constructor requires it to be non-const.

Reply:
That's right, `make_rch` doesn't have all the const/non-const combinations covered since it would be cumbersome.
You can change the `PmfSporadicTask` constructor to use const.  I tried it in branch `rch-experiment` of my fork and it compiled.